snforge_std/src/signature/key_pair.cairo

```diff
@@ -0,0 +1,18 @@
+#[derive(Copy, Drop)]
+struct KeyPair<SK, PK> {
+    secret_key: SK,
+    public_key: PK,
+}
+
+trait KeyPairTrait<SK, PK> {
+    fn generate() -> KeyPair<SK, PK>;
+    fn from_secret_key(secret_key: SK) -> KeyPair<SK, PK>;
+}
+
+trait SignerTrait<T, H, U> {
+    fn sign(self: T, message_hash: H) -> (U, U);
+}
+
+trait VerifierTrait<T, H, U> {
+    fn verify(self: T, message_hash: H, signature: (U, U)) -> bool;
+}
```

Comment:
Keep in mind that with the signature being just a single generic `U`, type annotations are needed:
```cairo
let (r, s): (felt252, felt252) = key_pair.sign(msg_hash);
```

instead of:
```cairo
let (r, s) = key_pair.sign(msg_hash);
```